packages/client/src/create-deployment.ts

```diff
@@ -90,27 +90,38 @@ export default function buildCreateDeployment() {
 
     let files;
 
-    if (clientOptions.archive === 'tgz') {
-      debug('Packing tarball');
-      const tarStream = tar
-        .pack(workPath, {
-          entries: fileList.map(file => relative(workPath, file)),
-        })
-        .pipe(createGzip());
-      const tarBuffer = await streamToBuffer(tarStream);
-      debug('Packed tarball');
-      files = new Map([
-        [
-          hash(tarBuffer),
-          {
-            names: [join(workPath, '.vercel/source.tgz')],
-            data: tarBuffer,
-            mode: 0o666,
-          },
-        ],
-      ]);
-    } else {
-      files = await hashes(fileList);
+    try {
+      if (clientOptions.archive === 'tgz') {
+        debug('Packing tarball');
+        const tarStream = tar
+          .pack(workPath, {
+            entries: fileList.map(file => relative(workPath, file)),
+          })
+          .pipe(createGzip());
+        const tarBuffer = await streamToBuffer(tarStream);
+        debug('Packed tarball');
+        files = new Map([
+          [
+            hash(tarBuffer),
+            {
+              names: [join(workPath, '.vercel/source.tgz')],
+              data: tarBuffer,
+              mode: 0o666,
+            },
+          ],
+        ]);
+      } else {
+        files = await hashes(fileList);
+      }
+    } catch (err: any) {
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** use `unknown` for error and type check it

Reply:
fbbab36453c76843972f601c26e07d29aa5630f0